### ISSUE TITLE
Higher-contrast focus outline color

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#8823]
### Is your proposal related to a problem?

Reviewing our focus outline with @wagtail/accessibility, based on reports from an accessibility specialist, we think our focus outline could be tweaked to have better contrast against a good range of Wagtail’s colors. In particular, it doesn’t score very well in [APCA](https://www.myndex.com/APCA/) against white.

### Describe the solution you'd like

We should choose a new focus outline color that scores better against white, and scores well against common colors of interactive elements. We’re in particular interested in a good APCA score, ideally rating "visible for icons" or higher. WCAG2 score is a bonus.

### Describe alternatives you've considered

We also discussed using a two-tone outline but decided this wasn’t worth the complexity for the time being.

### Additional context

Based on preliminary APCA guidance, our understanding is we need a minimum score of 30 against all of colors this will be displayed next to:

> Lc 30 • […] This is also the minimum for large/solid semantic & understandable non-text elements such as "mostly solid" icons or pictograms, no less than 10px in its smallest dimension.

A score of 45 wherever possible would be desirable as this is the recommendation for outlines (up to interpretation):

> Lc 45 • […] This is also the minimum for pictograms with fine details, or smaller outline icons, , no less than 4px in its smallest dimension.

---

- Current color: #ffbf47
- [APCA against white](https://www.myndex.com/APCA/?BG=ffffff&TXT=ffbf47&DEV=G4g&BUF=A22)
- [Contrast grid with our focus outline color against different backgrounds](https://contrast-grid.eightshapes.com/?version=1.1.0&background-colors=%23FFFFFF%2C%20white%0D%0A%232E1F5E%2C%20primary%0D%0A%23F6F6F8%2C%20grey-50%0D%0A%23E0E0E0%2C%20grey-100%0D%0A%23929292%2C%20grey-200%0D%0A%23004345%2Cteal-600%0D%0A%23005B5E%2Cteal-400%0D%0A%23007D7E%2Cteal-200%0D%0A%23F2FCFC%2Cteal-50%0D%0A%23FDE9E9%2C%20critical-50%0D%0A%23FD5765%2C%20critical-100%0D%0A%23CD4444%2C%20critical-200%0D%0A%23FAECD5%2C%20warning-50%0D%0A%23FFBF00%2C%20warning-100%0D%0A%23E0FBF4%2C%20positive-50%0D%0A%231B8666%2C%20positive-100%0D%0A%23E2F5FC%2C%20info-50%0D%0A%231F7E9A%2C%20info-100&foreground-colors=%23ffbf47%2Cfocus&es-color-form__tile-size=compact&es-color-form__show-contrast=aaa&es-color-form__show-contrast=aa&es-color-form__show-contrast=aa18&es-color-form__show-contrast=dnp).

**Comment by thibaudcolas** [comment]
Tagging this as a "good first issue" _for someone with accessibility expertise_. There is no particularly deep knowledge of Wagtail required to help with this, however this requires expert knowledge of accessibility and color contrast nonetheless.

If anyone is interested in picking this up, I’d recommend to start with a few suggestions of colors by using the Contrast Grid link I shared to compare them all against our palette.

**Comment by lb-** [comment]
@Akash-Kumar-Sen you may want to give this one a go - would be a great win for accessibility and now that you have dabbled a bit in outline colours maybe you want to take a look.

**Comment by Akash-Kumar-Sen** [comment]
'We should choose a new focus outline color that scores better against white' which page we are talking about here?

**Comment by lb-** [comment]
The goal is to find a colour that works better against most of the backgrounds (see the contrast grid). 

Contrast against white should not be too hard. But finding a contrast colour that also works well when focusing on the sidebar buttons (sitting on purple) and then inside messages is harder. 

I would say that the message backgrounds are less priority. Maybe we will end up with a seperate colour for outlines in messages.

**Comment by thibaudcolas** [comment]
I’ll remove the "good first issue" label from this based upon recent discussions. We want to switch to one of our light teal shades for the focus outline color, however that can only happen once we have fewer prominent buttons using similar shades, particularly the "filled" button variant. We also need to make sure the shade of teal we’re using for this has a 3:1 contrast ratio against white (not the case currently).

Here’s a prototype of applying this kind of focus style to our interactive elements:

![localhost_8000_admin_pages_81_edit_](https://user-images.githubusercontent.com/877585/184833364-eac39ee4-3fc7-4251-966e-5fa98b921fe2.png)

---

So this is blocked until we either change the color to a non-teal hue, or make enough changes to the UI that a teal hue works.

**Comment by lb-** [comment]
This is a helpful tool - let's you find colours that contrast against other colours but are similar to your preference. 

https://www.learnui.design/tools/accessible-color-generator.html

**Comment by lb-** [comment]
Also. I think this colour should be theme-able. In the colours.js (for tailwind) and exposed as a CSS custom property. This way custom colour themes can also provide a custom focus colour.

**Comment by nicklee** [comment]
If we want to stick with the green theme, we could go a little minty to ensure we're not too close to any existing colours, I've updated @thibaudcolas's grid here with the addition of the "new focus":

https://contrast-grid.eightshapes.com/?version=1.1.0&background-colors=%23FFFFFF%2C%20white%0D%0A%232E1F5E%2C%20primary%0D%0A%23F6F6F8%2C%20grey-50%0D%0A%23E0E0E0%2C%20grey-100%0D%0A%23929292%2C%20grey-200%0D%0A%23262626%2Cgrey-600%0D%0A%23004345%2Cteal-600%0D%0A%23005B5E%2Cteal-400%0D%0A%23007D7E%2Cteal-200%0D%0A%23F2FCFC%2Cteal-50%0D%0A%23FDE9E9%2C%20critical-50%0D%0A%23FD5765%2C%20critical-100%0D%0A%23CD4444%2C%20critical-200%0D%0A%23FAECD5%2C%20warning-50%0D%0A%23FFBF00%2C%20warning-100%0D%0A%23E0FBF4%2C%20positive-50%0D%0A%231B8666%2C%20positive-100%0D%0A%23E2F5FC%2C%20info-50%0D%0A%231F7E9A%2C%20info-100&foreground-colors=%23ffbf47%2Cfocus-current%0D%0A%2300B0B1%2Csecondary-100%0D%0A%23007D7E%2Csecondary-200%0D%0A%23005B5E%2Csecondary-400%0D%0A%2380D7D8%2Csecondary-75%0D%0A%23009072%2C%20new%20focus&es-color-form__tile-size=regular&es-color-form__show-contrast=aaa&es-color-form__show-contrast=aa&es-color-form__show-contrast=aa18&es-color-form__show-contrast=dnp

Screenshot showing how it works on both white and the darker wagtail palette: 


![Image](https://user-images.githubusercontent.com/298766/204571155-8682ebec-c3f7-46e7-9571-b4cf43ea9afb.png)

This is about as dark as I'd feel confident going without the colour becoming muddy, giving us a 4.1 contrast on white: https://colourcontrast.cc/ffffff/009072

**Comment by thibaudcolas** [comment]
Fixed in #9839
